include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -313,16 +313,17 @@ class format_parse : public format_base
      * \tparam option_t Must model seqan3::named_enumeration.
      * \param[out] value Stores the parsed value.
      * \param[in] in The input argument to be parsed.
-     * \returns seqan3::option_parse_result::error if `in` could not be found in the
-     *          seqan3::enumeration_names<option_t> map and otherwise seqan3::option_parse_result::success.
+     * \throws seqan3::user_input_error if `in` is not a key in seqan3::enumeration_names<option_t>.
+     * \returns seqan3::option_parse_result::success.
      */
     template <named_enumeration option_t>
     option_parse_result parse_option_value(option_t & value, std::string const & in)
     {
         auto map = seqan3::enumeration_names<option_t>;
 
         if (auto it = map.find(in); it == map.end())
-            return option_parse_result::error;
+            throw user_input_error{detail::to_string("You have choosen an invalid input value: ", in,
+                                                     " Please use one of: ", map | std::views::keys)};
```

Comment:
Would it look better to have a `.` before Please, i.e. `". Please use one of: "`? Not sure.